diff --git a/src/sshmasterconnection.cpp b/src/sshmasterconnection.cpp
index cd07806..5bf8a9e 100644
--- a/src/sshmasterconnection.cpp
+++ b/src/sshmasterconnection.cpp
@@ -157,11 +157,9 @@ SshMasterConnection::SshMasterConnection (QObject* parent, QString host, int por
         QString proxylogin, QString proxypassword, QString proxykey,
         bool proxyautologin, bool proxyKrbLogin ) : QThread ( parent )
 {
-#if defined ( Q_OS_DARWIN )
     // Mac OS X provides only 512KB stack space for secondary threads.
     // As we put a 512KB buffer on the stack later on, we need a bigger stack space.
     setStackSize (sizeof (char) * 1024 * 1024 * 2);
-#endif
     tcpProxySocket = NULL;
     tcpNetworkProxy = NULL;
     sshProxy= NULL;
