diff --git a/configure.ac b/configure.ac
index 381b621..481de31 100644
--- a/configure.ac
+++ b/configure.ac
@@ -17,7 +17,7 @@ AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_AUX_DIR([build-aux])
 AC_REQUIRE_AUX_FILE([tap-driver.sh])
 
-AM_INIT_AUTOMAKE(1.12 tar-ustar no-dist-gzip dist-bzip2 -Wno-portability)  dnl NB: Do not [quote] this parameter.
+AM_INIT_AUTOMAKE(1.12 tar-ustar no-dist-gzip dist-bzip2 subdir-objects -Wno-portability)  dnl NB: Do not [quote] this parameter.
 AM_MAINTAINER_MODE([enable])
 AM_SILENT_RULES([yes])
 
diff --git a/src/Makefile.am b/src/Makefile.am
index 7339058..308b26a 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -53,75 +53,74 @@ AM_CPPFLAGS += $(foreach d,$(sort $(dir $(libNetworkManager_la_SOURCES))),-I$(to
 
 noinst_LTLIBRARIES = \
 	libNetworkManager.la \
-	libnm-iface-helper.la \
-	libsystemd-dhcp.la
+	libnm-iface-helper.la
 
 ######################
 # libsystemd-dhcp
 ######################
 
-SYSTEMD_DHCP_CFLAGS = \
-	-I$(top_srcdir)/src/dhcp-manager/systemd-dhcp/src/systemd \
-	-I$(top_srcdir)/src/dhcp-manager/systemd-dhcp/src/libsystemd-network \
-	-I$(top_srcdir)/src/dhcp-manager/systemd-dhcp/src/shared \
-	-I$(top_srcdir)/src/dhcp-manager/systemd-dhcp
-
-libsystemd_dhcp_la_SOURCES = \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp-network.c \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp-packet.c \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp-internal.h \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp6-network.c \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp6-lease-internal.h \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp6-option.c \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/network-internal.c \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/sd-dhcp-lease.c \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/sd-dhcp-client.c \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp-option.c \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/network-internal.h \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/sd-dhcp6-lease.c \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp-protocol.h \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp6-internal.h \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp6-protocol.h \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp-lease-internal.h \
-	dhcp-manager/systemd-dhcp/src/libsystemd-network/sd-dhcp6-client.c \
-	dhcp-manager/systemd-dhcp/src/shared/async.h \
-	dhcp-manager/systemd-dhcp/src/shared/time-util.h \
-	dhcp-manager/systemd-dhcp/src/shared/siphash24.h \
-	dhcp-manager/systemd-dhcp/src/shared/time-util.c \
-	dhcp-manager/systemd-dhcp/src/shared/socket-util.h \
-	dhcp-manager/systemd-dhcp/src/shared/sparse-endian.h \
-	dhcp-manager/systemd-dhcp/src/shared/macro.h \
-	dhcp-manager/systemd-dhcp/src/shared/refcnt.h \
-	dhcp-manager/systemd-dhcp/src/shared/util.c \
-	dhcp-manager/systemd-dhcp/src/shared/in-addr-util.c \
-	dhcp-manager/systemd-dhcp/src/shared/siphash24.c \
-	dhcp-manager/systemd-dhcp/src/shared/util.h \
-	dhcp-manager/systemd-dhcp/src/shared/in-addr-util.h \
-	dhcp-manager/systemd-dhcp/src/shared/list.h \
-	dhcp-manager/systemd-dhcp/src/shared/fileio.h \
-	dhcp-manager/systemd-dhcp/src/shared/fileio.c \
-	dhcp-manager/systemd-dhcp/src/shared/strv.h \
-	dhcp-manager/systemd-dhcp/src/shared/strv.c \
-	dhcp-manager/systemd-dhcp/src/shared/unaligned.h \
-	dhcp-manager/systemd-dhcp/src/shared/utf8.h \
-	dhcp-manager/systemd-dhcp/src/shared/utf8.c \
-	dhcp-manager/systemd-dhcp/src/systemd/sd-dhcp-lease.h \
-	dhcp-manager/systemd-dhcp/src/systemd/sd-dhcp-client.h \
-	dhcp-manager/systemd-dhcp/src/systemd/sd-id128.h \
-	dhcp-manager/systemd-dhcp/src/systemd/sd-dhcp6-lease.h \
-	dhcp-manager/systemd-dhcp/src/systemd/sd-dhcp6-client.h \
-	dhcp-manager/systemd-dhcp/src/systemd/sd-event.h \
-	dhcp-manager/systemd-dhcp/src/systemd/_sd-common.h \
-	dhcp-manager/systemd-dhcp/nm-sd-adapt.h \
-	dhcp-manager/systemd-dhcp/nm-sd-adapt.c
-
-libsystemd_dhcp_la_CPPFLAGS = \
-	-I$(top_srcdir)/include \
-	$(SYSTEMD_DHCP_CFLAGS) \
-	$(GLIB_CFLAGS)
-
-libsystemd_dhcp_la_LIBADD = \
-	$(GLIB_LIBS)
+#SYSTEMD_DHCP_CFLAGS = \
+#	-I$(top_srcdir)/src/dhcp-manager/systemd-dhcp/src/systemd \
+#	-I$(top_srcdir)/src/dhcp-manager/systemd-dhcp/src/libsystemd-network \
+#	-I$(top_srcdir)/src/dhcp-manager/systemd-dhcp/src/shared \
+#	-I$(top_srcdir)/src/dhcp-manager/systemd-dhcp
+
+#libsystemd_dhcp_la_SOURCES = \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp-network.c \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp-packet.c \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp-internal.h \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp6-network.c \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp6-lease-internal.h \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp6-option.c \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/network-internal.c \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/sd-dhcp-lease.c \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/sd-dhcp-client.c \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp-option.c \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/network-internal.h \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/sd-dhcp6-lease.c \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp-protocol.h \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp6-internal.h \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp6-protocol.h \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/dhcp-lease-internal.h \
+#	dhcp-manager/systemd-dhcp/src/libsystemd-network/sd-dhcp6-client.c \
+#	dhcp-manager/systemd-dhcp/src/shared/async.h \
+#	dhcp-manager/systemd-dhcp/src/shared/time-util.h \
+#	dhcp-manager/systemd-dhcp/src/shared/siphash24.h \
+#	dhcp-manager/systemd-dhcp/src/shared/time-util.c \
+#	dhcp-manager/systemd-dhcp/src/shared/socket-util.h \
+#	dhcp-manager/systemd-dhcp/src/shared/sparse-endian.h \
+#	dhcp-manager/systemd-dhcp/src/shared/macro.h \
+#	dhcp-manager/systemd-dhcp/src/shared/refcnt.h \
+#	dhcp-manager/systemd-dhcp/src/shared/util.c \
+#	dhcp-manager/systemd-dhcp/src/shared/in-addr-util.c \
+#	dhcp-manager/systemd-dhcp/src/shared/siphash24.c \
+#	dhcp-manager/systemd-dhcp/src/shared/util.h \
+#	dhcp-manager/systemd-dhcp/src/shared/in-addr-util.h \
+#	dhcp-manager/systemd-dhcp/src/shared/list.h \
+#	dhcp-manager/systemd-dhcp/src/shared/fileio.h \
+#	dhcp-manager/systemd-dhcp/src/shared/fileio.c \
+#	dhcp-manager/systemd-dhcp/src/shared/strv.h \
+#	dhcp-manager/systemd-dhcp/src/shared/strv.c \
+#	dhcp-manager/systemd-dhcp/src/shared/unaligned.h \
+#	dhcp-manager/systemd-dhcp/src/shared/utf8.h \
+#	dhcp-manager/systemd-dhcp/src/shared/utf8.c \
+#	dhcp-manager/systemd-dhcp/src/systemd/sd-dhcp-lease.h \
+#	dhcp-manager/systemd-dhcp/src/systemd/sd-dhcp-client.h \
+#	dhcp-manager/systemd-dhcp/src/systemd/sd-id128.h \
+#	dhcp-manager/systemd-dhcp/src/systemd/sd-dhcp6-lease.h \
+#	dhcp-manager/systemd-dhcp/src/systemd/sd-dhcp6-client.h \
+#	dhcp-manager/systemd-dhcp/src/systemd/sd-event.h \
+#	dhcp-manager/systemd-dhcp/src/systemd/_sd-common.h \
+#	dhcp-manager/systemd-dhcp/nm-sd-adapt.h \
+#	dhcp-manager/systemd-dhcp/nm-sd-adapt.c
+
+#libsystemd_dhcp_la_CPPFLAGS = \
+#	-I$(top_srcdir)/include \
+#	$(SYSTEMD_DHCP_CFLAGS) \
+#	$(GLIB_CFLAGS)
+
+#libsystemd_dhcp_la_LIBADD = \
+#	$(GLIB_LIBS)
 
 ###########################################
 # NetworkManager
@@ -166,14 +165,12 @@ nm_device_headers = \
 nm_dhcp_client_sources = \
 	dhcp-manager/nm-dhcp-dhclient.c \
 	dhcp-manager/nm-dhcp-dhclient-utils.c \
-	dhcp-manager/nm-dhcp-dhcpcd.c \
-	dhcp-manager/nm-dhcp-systemd.c
+	dhcp-manager/nm-dhcp-dhcpcd.c 
 
 nm_dhcp_client_headers = \
 	dhcp-manager/nm-dhcp-dhclient.h \
 	dhcp-manager/nm-dhcp-dhclient-utils.h \
-	dhcp-manager/nm-dhcp-dhcpcd.h \
-	dhcp-manager/nm-dhcp-systemd.h
+	dhcp-manager/nm-dhcp-dhcpcd.h
 
 nm_sources = \
 	$(nm_device_headers) \
@@ -345,7 +342,7 @@ nm_sources = \
 	NetworkManagerUtils.h
 
 if SESSION_TRACKING_SYSTEMD
-nm_sources += nm-session-monitor-systemd.c
+#nm_sources += nm-session-monitor-systemd.c
 else
 if SESSION_TRACKING_CK
 nm_sources += nm-session-monitor-ck.c
@@ -355,7 +352,7 @@ endif
 endif
 
 if SUSPEND_RESUME_SYSTEMD
-nm_sources += nm-sleep-monitor-systemd.c
+#nm_sources += nm-sleep-monitor-systemd.c
 else
 # UPower suspend/resume used whenever systemd is not enabled
 nm_sources += nm-sleep-monitor-upower.c
@@ -453,7 +450,6 @@ libNetworkManager_la_SOURCES = \
 
 libNetworkManager_la_LIBADD = \
 	$(top_builddir)/libnm-core/libnm-core.la \
-	libsystemd-dhcp.la \
 	$(DBUS_LIBS) \
 	$(GLIB_LIBS) \
 	$(GUDEV_LIBS) \
@@ -525,7 +521,6 @@ endif
 
 libnm_iface_helper_la_LIBADD = \
 	$(top_builddir)/libnm-core/libnm-core.la \
-	libsystemd-dhcp.la \
 	$(DBUS_LIBS) \
 	$(GLIB_LIBS) \
 	$(GUDEV_LIBS) \
@@ -537,15 +532,12 @@ libnm_iface_helper_la_LIBADD = \
 libexec_PROGRAMS = nm-iface-helper
 
 nm_iface_helper_SOURCES = \
-	dhcp-manager/nm-dhcp-systemd.h \
-	dhcp-manager/nm-dhcp-systemd.c \
 	nm-iface-helper.c \
 	main-utils.c \
 	main-utils.h
 
 nm_iface_helper_LDADD = \
 	$(top_builddir)/libnm-core/libnm-core.la \
-	libsystemd-dhcp.la \
 	libnm-iface-helper.la \
 	$(DBUS_LIBS) \
 	$(GLIB_LIBS) \
