diff --git a/client/X11/xf_window.c b/client/X11/xf_window.c
index a24ac71..f2d09ff 100644
--- a/client/X11/xf_window.c
+++ b/client/X11/xf_window.c
@@ -146,7 +146,7 @@ void xf_SetWindowFullscreen(xfContext* xfc, xfWindow* window, BOOL fullscreen)
 
 		xf_SetWindowDecorations(xfc, window, FALSE);
 
-		XMoveResizeWindow(xfc->display, window->handle, settings->DesktopPosX, settings->DesktopPosY, window->width, window->height);
+		XMoveResizeWindow(xfc->display, window->handle, 0, 0, window->width, window->height);
                 XMapRaised(xfc->display, window->handle);
 
 		window->fullscreen = TRUE;
@@ -437,7 +437,7 @@ xfWindow* xf_CreateDesktopWindow(xfContext* xfc, char* name, int width, int heig
 		}
 		else if (settings->DesktopPosX || settings->DesktopPosY)
 		{
-			XMoveWindow(xfc->display, window->handle, settings->DesktopPosX, settings->DesktopPosY);
+			XMoveWindow(xfc->display, window->handle, 0, 0);
 		}
 	}
 
