diff --git a/winpr/tools/makecert/cli/CMakeLists.txt b/winpr/tools/makecert/cli/CMakeLists.txt
index f175e72..031e5e1 100644
--- a/winpr/tools/makecert/cli/CMakeLists.txt
+++ b/winpr/tools/makecert/cli/CMakeLists.txt
@@ -45,6 +45,9 @@ set(${MODULE_PREFIX}_LIBS winpr-makecert-tool)
 
 target_link_libraries(${MODULE_NAME} ${${MODULE_PREFIX}_LIBS})
 
+set_target_properties(${MODULE_NAME} PROPERTIES 
+	BUILD_WITH_INSTALL_RPATH 1)
+
 set_property(TARGET ${MODULE_NAME} PROPERTY FOLDER "WinPR/Tools")
 
 install(TARGETS ${MODULE_NAME} DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT tools EXPORT WinPRTargets)
