diff --git a/src/util/u_atomic.h b/src/util/u_atomic.h
index e38395a..5310d67 100644
--- a/src/util/u_atomic.h
+++ b/src/util/u_atomic.h
@@ -38,14 +38,14 @@
 
 #define p_atomic_set(_v, _i) (*(_v) = (_i))
 #define p_atomic_read(_v) (*(_v))
-#define p_atomic_dec_zero(v) (__sync_sub_and_fetch((v), 1) == 0)
-#define p_atomic_inc(v) (void) __sync_add_and_fetch((v), 1)
-#define p_atomic_dec(v) (void) __sync_sub_and_fetch((v), 1)
-#define p_atomic_add(v, i) (void) __sync_add_and_fetch((v), (i))
-#define p_atomic_inc_return(v) __sync_add_and_fetch((v), 1)
-#define p_atomic_dec_return(v) __sync_sub_and_fetch((v), 1)
+#define p_atomic_dec_zero(v) (__sync_sub_and_fetch((unsigned short int *)(v), 1) == 0)
+#define p_atomic_inc(v) (void) __sync_add_and_fetch((unsigned short int *)(v), 1)
+#define p_atomic_dec(v) (void) __sync_sub_and_fetch((unsigned short int *)(v), 1)
+#define p_atomic_add(v, i) (void) __sync_add_and_fetch((unsigned short int *)(v), (i))
+#define p_atomic_inc_return(v) __sync_add_and_fetch((unsigned short int *)(v), 1)
+#define p_atomic_dec_return(v) __sync_sub_and_fetch((unsigned short int *)(v), 1)
 #define p_atomic_cmpxchg(v, old, _new) \
-   __sync_val_compare_and_swap((v), (old), (_new))
+   __sync_val_compare_and_swap((unsigned short int *)(v), (old), (_new))
 
 #endif
 
